app/models/group.rb

```diff
@@ -1,3 +1,8 @@
 class Group < ApplicationRecord
+  belongs_to :teacher
   has_many :students_groups
```

Comment:
add dependent: :destroy to has_many :students_groups

Reply:
ok, done